code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -1336,3 +1336,837 @@
+		S.encryption_key = null
+		to_chat(src, SPAN_NOTICE("Your Hivenet encryption key has been reset."))
+		return
+	S.encryption_key = input(src, "Enter a new encryption key for Hivenet messages.", "Hivenet Decryption") as text
+	to_chat(src, SPAN_NOTICE("Your Hivenet encryption key has been set to [S.encryption_key]."))
+
+/mob/living/carbon/human/proc/hivenet_decrypt()
+	set name = "Decrypt Hivenet Key"
+	set desc = "Attempt to retrieve an in-use Hivenet encryption key."
+	set category = "Hivenet"
+
+	var/decryptchance = 5
+	var/obj/item/organ/internal/vaurca/neuralsocket/host = src.internal_organs_by_name[BP_NEURAL_SOCKET]
+	if(src.stat != CONSCIOUS)
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
+		return
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before attempting Hivenet decryption!"))
+	if(istype(src.origin, /singleton/origin_item/origin/mouv_b))
+		decryptchance = 15
+	for(var/mob/living/carbon/human/player in player_list)
+		if(isvaurca(player) && player.internal_organs_by_name[BP_NEURAL_SOCKET])
+			var/obj/item/organ/internal/vaurca/neuralsocket/S = player.internal_organs_by_name[BP_NEURAL_SOCKET]
+			if(S.encryption_key)
+				if(prob(decryptchance))
+					to_chat(src, SPAN_NOTICE("You are able to retrieve a Hivenet encryption key. The key is \"[S.encryption_key]\""))
+					host.last_action = world.time + 5 MINUTES
+					return
+	host.last_action = world.time + 5 MINUTES
+	to_chat(src, SPAN_NOTICE("You are unable to retrieve any Hivenet encryption keys."))
+
+/mob/living/carbon/human/proc/hivenet_camera()
+	set name = "Access Hivenet User Senses"
+	set desc = "Attempt to look through the eyes of another Vaurca as if they were a camera. Vaurcae of your own Hive have no choice unless they have countermeasures, whereas Vaurca of other Hives can choose to allow you access. A Mouv Ta may attempt to force this, but it would be a serious diplomatic breach."
+	set category = "Hivenet"
+
+	var/list/available_vaurca = list()
+	var/list/fullname = splittext(src.name, " ")
+	var/surname = fullname[2]
+	var/obj/item/organ/internal/vaurca/neuralsocket/host = src.internal_organs_by_name[BP_NEURAL_SOCKET]
+	if(src.stat != CONSCIOUS)
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
+		return
+	if(stat!=CONSCIOUS)
+		remoteview_target = null
+		reset_view(0)
+		return
+	if(client.eye != client.mob)
+		remoteview_target = null
+		reset_view(0)
+		return
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before attempting another Hivenet action!"))
```

Comment:
this needs to return, and should be at the start of the proc, right below the stat check